### DISCUSSION TITLE
Binary Data Example

### DISCUSSION THREAD

**Discussion by geyang** [prisma/prisma#13637] [Q&A]
Hi, the following query returns a type checking error:

The schema is
```prisma
model Text {
  id     String  @id @default(auto()) @map("_id") @db.ObjectId
  text   String?
  deltas Bytes[] //@db.BinData
}
```

```javascript
changes = original.map(s => new Binary(toUint8Array(s), BSON_BINARY_SUBTYPE_BYTE_ARRAY))
file = await context.prisma.file.create({
          data: text
            ? {text, changes}
            : {deltas}
        })
```

```error
Argument deltas: Got invalid value 
[
  {
    sub_type: 2,
    buffer: Buffer(3),
    position: 25
  },
  {
    sub_type: 2,
    buffer: Buffer(3),
    position: 22
  },
  {
    sub_type: 2,
    buffer: Buffer(3),
    position: 24
  }
]
on prisma.createOneFile. Provided List<Json>, expected FileCreatedeltasInput or Bytes:
type FileCreatedeltasInput {
  set: List<Bytes>
}
type FileCreatedeltasInput {
  set: List<Bytes>
}
```

**Comment by austincrim** [comment]
Hey @geyang 👋,

What is the `Binary` type you are instantiating inside the `map` function? The `Bytes` type [typically accepts `Buffers`](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields#working-with-bytes).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
